datastore/gcloud/aio/datastore/filter.py

```diff
@@ -0,0 +1,115 @@
+from typing import Any
+from typing import Dict
+from typing import List
+
+from gcloud.aio.datastore.constants import CompositeFilterOperator
+from gcloud.aio.datastore.constants import PropertyFilterOperator
+from gcloud.aio.datastore.value import Value
+
+
+class BaseFilter:
+    json_key: str
+
+    # pylint: disable=unused-argument
+    def __new__(cls, *args, **kwargs):
+        if cls is BaseFilter:
+            raise TypeError('BaseFilter may not be instantiated')
+        return object.__new__(cls)
```

Comment:
Use `super()` rather than `object` -- always good to use the builtins, in case we have `BaseFilter` inherit from something else in the future.